### PR TITLE
Add more documentation

### PR DESCRIPTION
**Pull request by scanon**
Should we add documentation on EE2 deployment here as well or anything about how to get variables from EE2 into this? or maybe in a separate issue?

### CODE REVIEW

**Review by Tianhao-Gu — APPROVED** [review]
Thanks for the doc. Container version works for me locally.

File: README.md [hunk 1]
```diff
@@ -2,6 +2,87 @@
...
+# Set tokens and URL
+export KB_AUTH_TOKEN="xxxxxxxxx"
+export KB_ADMIN_AUTH_TOKEN="xxxxxxxxxxxx"
+export KB_BASE_URL=https://ci.kbase.us/services
+
+# Optional
+export KB_REF_DATA=/path/to/local/refdata
+
+# Launch the callback server
+python -m JobRunner.callback
```
[the remark this answers is not in the document]
Comment (bio-boris) [reply README.md:32]: @Tianhao-Gu here how I set it it for me (You can use your podman or rancher desktop socket or colima socket)
export DOCKER_HOST=unix://$HOME/.colima/docker.sock

File: README.md [hunk 2]
```diff
@@ -2,6 +2,86 @@
...
+Here is a quick start guide for running test for the Job Runner code.
+See the SDK guide for information about running test of SDK apps.
+
+```
+pip install -r requirements.txt -r requirements-dev.txt
+
+# Set the DOCKER_HOST if this doesn't work out of the box
+export DOCKER_HOST=unix://$HOME/.docker/run/docker.sock
+
+# Set tokens and URL
```
Comment (bio-boris) [inline README.md:73]:
```suggestion
# Be sure to set both tokens and the KB_BASE_URL
```

File: README.md [hunk 3]
```diff
@@ -2,6 +2,86 @@
...
+### Container Launch
+
+The callback server can also be launched via a container.  This may be useful
+where the local python and what works with the JobRunner are incompatible.
+Note that the JOB_DIR needs to be accessible by docker.  Also you must pass through
+the docker socket to the container.  The path to the socket can vary depending on
+the container runtime and how the container runtime is configured.
+
+```
+export JOB_DIR=/full/path/to/work/area
```
Comment (bio-boris) [inline README.md:44]: I had to chmod -R 777 this directory for callbacks to work, should we add this?

**Review by bio-boris — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -2,6 +2,86 @@
...
+
+# Set the DOCKER_HOST if this doesn't work out of the box
+export DOCKER_HOST=unix://$HOME/.docker/run/docker.sock
+
+# Set tokens and URL
+export KB_AUTH_TOKEN="xxxxxxxxx"
+export KB_ADMIN_AUTH_TOKEN="xxxxxxxxxxxx"
+export KB_BASE_URL=https://ci.kbase.us/services
+
+# Set ref data to an area acceessible by Docker
```
Comment (bio-boris) [inline README.md:78]:
```suggestion
# Set ref data to an area accessible by Docker
```

File: README.md [hunk 5]
```diff
@@ -2,6 +2,86 @@
...
+export KB_REF_DATA=/path/to/local/refdata
+
+make mock
+make testimage
+make test
```
Comment (bio-boris) [inline README.md:83]: RE Make test

* Nose is not installed
* coverage is not installed
* Won't run on python 3.11 (                         ^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'collections' has no attribute 'Callable')

Comment (scanon) [reply README.md:84]: If you install requirements-dev.txt it will add those.  I will make a comment that this is currently working against Python 3.9

**Review by bio-boris — APPROVED** [review]

**Pull request merged**